src/layers/core/point-cloud-layer/point-cloud-layer.js

```diff
@@ -0,0 +1,140 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {Layer} from '../../../lib';
+import {assembleShaders} from '../../../shader-utils';
+import {GL, Model, Geometry} from 'luma.gl';
+import {readFileSync} from 'fs';
+import {join} from 'path';
+
+const DEFAULT_COLOR = [0, 0, 0, 255];
+
+const defaultProps = {
+  getPosition: x => x.position,
+  getNormal: x => x.normal,
+  getColor: x => x.color || DEFAULT_COLOR,
+  radius: 10,  //  point radius in pixels
+  lightSettings: {
+    lightsPosition: [0, 0, 5000, -1000, 1000, 8000, 5000, -5000, 1000],
+    ambientRatio: 0.2,
+    diffuseRatio: 0.6,
+    specularRatio: 0.8,
+    lightsStrength: [1.0, 0.0, 0.8, 0.0, 0.4, 0.0],
+    numberOfLights: 3
+  }
+};
+
+export default class ScatterplotLayer extends Layer {
+  getShaders(id) {
+    return {
+      vs: readFileSync(join(__dirname, './point-cloud-layer-vertex.glsl'), 'utf8'),
+      fs: readFileSync(join(__dirname, './point-cloud-layer-fragment.glsl'), 'utf8'),
+      modules: ['lighting']
+    };
+  }
+
+  initializeState() {
+    const {gl} = this.context;
+    this.setState({model: this._getModel(gl)});
+
+    /* eslint-disable max-len */
+    this.state.attributeManager.addInstanced({
+      instancePositions: {size: 3, accessor: 'getPosition', update: this.calculateInstancePositions},
+      instanceNormals: {size: 3, accessor: 'getNormal', defaultValue: 1, update: this.calculateInstanceNormals},
+      instanceColors: {size: 4, type: GL.UNSIGNED_BYTE, accessor: 'getColor', update: this.calculateInstanceColors}
+    });
+    /* eslint-enable max-len */
+  }
+
+  draw({uniforms}) {
+    const {radius, lightSettings} = this.props;
+    const {viewport: {width, height}} = this.context;
+    this.state.model.render(Object.assign({}, uniforms, {
+      radius,
+      screenSize: [width, height]
+    }, lightSettings));
+  }
+
+  _getModel(gl) {
+    // a triangle that minimally cover the unit circle
```

Comment:
> Rendering triangles is about 1/6 faster than squares.

@pessimistress Interesting! Out of curiosity, did you try different `radius` values? Intuitively it would make sense if: 
* small on-screen circles (that rasterize into few frags) benefitted from a primitive with less verts to process since the number of wasted fragment shader invocations will be small (this is consistent with your measurements),
* big on-screen circles that rasterize into a lot of frags benefitted from a primitive with less wasted frags. 

For the point cloud layer `radius` would typically be small, then a single triangle should be the right choice here.

But for `ScatterplotLayer` it might make sense to make another measurement before changing to a triangle primitive.